### PR TITLE
Pr/540 Added IPixel.PackFromBgra32 definition/implementations.

### PR DESCRIPTION
**Pull request by woutware**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [ ] I have provided test coverage for my change (where applicable)

### CODE REVIEW

File: src/ImageSharp/PixelFormats/Alpha8.cs [hunk 1]
```diff
@@ -97,6 +97,19 @@ public void PackFromRgba32(Rgba32 source)
             this.PackedValue = source.A;
         }
 
+        /// <inheritdoc/>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void PackFromArgb32(Argb32 source) {
```
Comment (iamcarbon) [inline src/ImageSharp/PixelFormats/Alpha8.cs]: Braces should be on their own line for consistency with the rest of the code base.

**Comment by woutware** [comment]
The changes are fairly straightforward and mostly copy/pasted from the Rgba32, since the added conversions are also from 4-byte pixel formats the code is mostly identical, just setting rgba fields or using the From/ToVector4 stuff. So I'd say it's not a high risk change. But I'm in no hurry with the merge. I can write some tests early next week (very maybe tomorrow).

File: src/ImageSharp/PixelFormats/Argb32.cs [hunk 2]
```diff
@@ -93,9 +102,11 @@ public Argb32(float r, float g, float b, float a = 1)
         /// <param name="vector">
         /// The vector containing the components for the packed vector.
         /// </param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public Argb32(Vector3 vector)
```
Comment (iamcarbon) [inline src/ImageSharp/PixelFormats/Argb32.cs]: We can also update the constructor to take a ref:

```
public Argb32(ref Vector3 vector)
```

File: src/ImageSharp/PixelFormats/Argb32.cs [hunk 3]
```diff
@@ -115,84 +128,36 @@ public Argb32(Vector4 vector)
...
 
         /// <summary>
-        /// Gets or sets the green component.
+        /// Gets or sets the packed representation of the Argb32 struct.
         /// </summary>
-        public byte G
+        public uint Argb
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get
```
Comment (iamcarbon) [inline src/ImageSharp/PixelFormats/Argb32.cs]: These format nicely on single lines with rocketships.
```
get => return Unsafe.As<Argb32, uint>(ref this);
```

File: src/ImageSharp/PixelFormats/Argb32.cs [hunk 4]
```diff
@@ -284,6 +269,13 @@ public void ToRgba32(ref Rgba32 dest)
             dest.A = this.A;
         }
 
+        /// <inheritdoc />
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void ToArgb32(ref Argb32 dest)
+        {
+            dest = this;
```
Comment (iamcarbon) [inline src/ImageSharp/PixelFormats/Argb32.cs]: NICE! 👍

File: src/ImageSharp/PixelFormats/Bgr24.cs [hunk 5]
```diff
@@ -82,6 +82,23 @@ public void PackFromRgba32(Rgba32 source)
             this = source.Bgr;
         }
 
+        /// <inheritdoc/>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void PackFromArgb32(Argb32 source) {
```
Comment (iamcarbon) [inline src/ImageSharp/PixelFormats/Bgr24.cs]: Put brace on new line.

File: src/ImageSharp/PixelFormats/Bgr24.cs [hunk 6]
```diff
@@ -131,6 +148,15 @@ public void ToRgba32(ref Rgba32 dest)
             dest.A = 255;
         }
 
+        /// <inheritdoc/>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void ToArgb32(ref Argb32 dest) {
```
Comment (iamcarbon) [inline src/ImageSharp/PixelFormats/Bgr24.cs]: Put brace on new line.

File: src/ImageSharp/PixelFormats/Bgr565.cs [hunk 7]
```diff
@@ -141,6 +154,16 @@ public void ToRgba32(ref Rgba32 dest)
             dest.A = (byte)MathF.Round(vector.W);
         }
 
+        /// <inheritdoc />
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void ToArgb32(ref Argb32 dest) {
```
Comment (iamcarbon) [inline src/ImageSharp/PixelFormats/Bgr565.cs]: Put brace on new line.

File: src/ImageSharp/PixelFormats/Bgra4444.cs [hunk 8]
```diff
@@ -132,6 +146,16 @@ public void ToRgba32(ref Rgba32 dest)
             dest.A = (byte)vector.W;
         }
 
+        /// <inheritdoc />
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void ToArgb32(ref Argb32 dest) {
```
Comment (iamcarbon) [inline src/ImageSharp/PixelFormats/Bgra4444.cs]: Put brace on new line. I'll stop commenting on the braces!

**Comment by JimBobSquarePants** [comment]
I've just pushed a bunch of additional tests. This should be ready for review now.

**Pull request merged**